Adding French translations

LyesSaadi opened the pull request:
> This is a WIP: everything is translated, but I still have to test things!

Hi!

So I translated the entirety of Setzer...

I ~~surely~~ might have done some spelling/grammar errors (I caught one just earlier), so I still have to review all of my translations again!

I also still need to be sure that what I translated is coherent with the context in which the string appeared.

PS: I never used BibTeX (only LaTeX), so I might have mistranslated some strings. I did my best to understand all terms ~~by doing some web searches~~, but I'll be grateful if some fellow French speaker that used it could maybe review what I translated to be 100% sure that no error slipped through !

cvfosammmm commented:
Thanks a lot for this effort.

LyesSaadi commented:
Oh, I was about to point out that I've made mnemonics translatable but I see that you already did it (I've missed one though :stuck_out_tongue: !) !

cvfosammmm commented:
Ok thanks. It would be nice btw. if you could open separate PRs for these additional fixes. It makes reviewing easier and I can merge them faster.

LyesSaadi commented:
I will, I won't do any change in the code (or I'll open another PR) ! I'll just concentrate on making this merge-able (since changes in master were made) and fixing some grammar/semantic errors !

LyesSaadi commented:
Hi!

Sorry for this late answer, I had urgent things to do before.

As you asked me to do, I removed all my local code changes, limiting it only to translation. And I based this PR on the latest commit.

And I fixed some errors, some might still have slipped through.

cvfosammmm commented:
Thanks. Is it still WIP?

LyesSaadi commented:
I just forgot the edit the title! It no longer is WIP.

cvfosammmm commented:
- Is there a shorter word for "warning" and "badbox" maybe?

LyesSaadi commented:
For "warning" there's "alerte". For "badbox", "débordement" isn't a good translation, but I cannot find any better... A good translation would be "Problème de mise en page" bu...

LyesSaadi commented:
Wow, so, I inadvertently closed the PR by accident, sorry for that!

LyesSaadi commented:
For "warning" there's "alerte". For "badbox", "débordement" isn't a good translation, but I cannot find any better... A good translation would be "problème de mise en page" but it is way too long...

But after some thinking, I'll keep the English term badbox like in other translations :)!

cvfosammmm commented:
"alerte" and "badbox" are fine.

dotlambda commented on po/fr.po:
```diff
@@ -0,0 +1,2453 @@
+
+#: data/org.cvfosammmm.Setzer.appdata.xml.in:34
+msgid "Document creation wizard"
+msgstr "Assistant de création de document"
```
```suggestion
msgstr "Assistant de création de documents"
```

dotlambda commented on po/fr.po:
```diff
@@ -0,0 +1,2453 @@
+
+#: data/org.cvfosammmm.Setzer.appdata.xml.in:21
+msgid "Arguably the best .pdf viewer of any LaTeX editor"
+msgstr "Sûrement le meilleur visionneur PDF de n'importe quel éditeur LaTeX"
```
I think "sûrement" is a bad translation of "arguably".
```suggestion
msgstr "Peut-être la meilleure visionneuse PDF parmi les éditeurs LaTeX"
```

LyesSaadi replied:
For this, now that I'm thinking of it "Sans doute" might be even better.
Like in: "Sans doute le meilleur visionneur PDF des éditeurs LaTeX" (Litteral translation to English => "Without doubt, the best PDF viewer of LaTeX editors").

What do you think of it?

PS: PDF is Masculine in French, as in "Le Document PDF".

dotlambda replied:
I still think "sans doute" is stronger than "arguably". But that's up to @cvfosammmm to decide.

Ah, "visionneur" seems to be fine: https://help.gnome.org/users/evince/stable/index.html.fr
I just found "visionneuse" on e.g. https://get.adobe.com/fr/reader/otherversions/.
It's independent from the gender of "PDF" btw.

LyesSaadi replied:
>  It's independent from the gender of "PDF" btw.
Yes, indeed, I confused the noun "visionneur" and adjective "visionnaire" ! And I looked more into it, and it seems that both are true, but masculine is definitely the more used one in my daily life:
- Larousse uses the feminine: https://www.larousse.fr/dictionnaires/francais/visionneuse/82203
- L'internaute uses the masculine: https://www.linternaute.fr/dictionnaire/fr/definition/visionneur/ (with a great example !)

cvfosammmm replied:
"Sans doute": I think that's too strong :), "Peut-être" seems fine. But I don't speak french.

LyesSaadi commented on po/fr.po:
```diff
@@ -0,0 +1,2453 @@
+
+#: setzer/dialogs/add_remove_packages/add_remove_packages.py:39
+msgid "mathematics package from AMS."
+msgstr "Paquets mathématiques de l'AMS."
```
Yes indeed ! Sorry for that !

LyesSaadi replied:
I just discovered that I've had already taken care of that on my local copy :sweat_smile: !

LyesSaadi commented on po/fr.po:
```diff
@@ -0,0 +1,2453 @@
+#: setzer/dialogs/add_remove_packages/add_remove_packages.py:42
+#: setzer/dialogs/document_wizard/pages/page_general_settings.py:164
+msgid "create a glossary for your document."
+msgstr "Crée un glossaire pour votre document."
```
So, I see that you prefer using Infinitive instead or the Declarative style of the Present. When translating the software, I wasn't sure which to use (both are valid and commonly used in different software). I think I chose at the time Infinitive seeing the rest of the translation, and some must have slipped. So, good catch :wink: !

LyesSaadi commented on po/fr.po:
```diff
@@ -0,0 +1,2453 @@
+#: setzer/dialogs/add_remove_packages/add_remove_packages.py:48
+#: setzer/dialogs/document_wizard/pages/page_general_settings.py:157
+msgid "contains symbols to be used in textmode."
+msgstr "Contiens des symboles utilisable en mode texte."
```
Good catch ! Thank you :grin: !

LyesSaadi commented on po/fr.po:
```diff
@@ -0,0 +1,2453 @@
+"define theorem environments (like \"definition\", \"lemma\", ...) with "
+"custom styling."
+msgstr ""
+"Définit l'environnement de théorème (comme \"definition\", \"lemma\", ...) "
```
Good catch :smiley: !

LyesSaadi commented on po/fr.po:
```diff
@@ -0,0 +1,2453 @@
+
+#: setzer/workspace/sidebar/sidebar.py:49
+msgid "Misc. Math"
+msgstr "Divers symboles mathématiques"
```
Both are valid ! But, I do prefer "Divers symboles" over "Symboles divers".

LyesSaadi commented:
Thank you a lot for this review :heart: ! This was my first try at translating a software, and, it's quite hard :stuck_out_tongue:. I'll reread my entire translation to fix any other grammar errors I might catch.

Unfortunately, until the 3rd of September, I'm quite busy, so I don't think I'll have a good translation until then.

I really wanted a French-speaking reviewer to review the translation, and I can't thank you more for finding those errors :grin: !

LyesSaadi commented on po/fr.po:
```diff
@@ -0,0 +1,2453 @@
+
+#: setzer/dialogs/add_remove_packages/add_remove_packages.py:41
+msgid "math theorems setting package from AMS."
+msgstr "Paquet de configuration pour les théorèmes mathématiques de l'AMS."
```
This string was removed in Setzer!

oscfdezdz commented:
Hi @LyesSaadi, is this still WIP? We are integrating Weblate and would like to add your contribution through it, thank you!
